### ISSUE TITLE
Detection of embed in Teams: Authentication pop-up window stays open

### ISSUE THREAD

**Issue by marekcis** [OfficeDev/microsoft-teams-library-js#872]
Hi,
as we are developing integration with Teams SDK for quite some time now, we have come across some behaviour related to authentication pop-up. Let me illustrate:
1. user clicks on Sign-in that opens authentication pop-up
2. in pop-up, we offer user to select login using SSO (e.g. Google)
3. user inputs invalid password and is taken somewhere else to reset password, or the input email needs SSO instead of password
4. the process succeeds and user stays in pop-up
5. after successful login, we redirect user to our app **but in the pop-up window**, because we lost track that we are in Teams integration flow

This issue was discussed also in [this issue](https://github.com/OfficeDev/microsoft-teams-library-js/issues/469#issuecomment-973054576) but was closed with solutions that are not applicable in our case since the URL content search parameter can get lost along the many redirects between prompt and successful login. And other detection methods are specific to being embedded in MS Teams, not about being opened by MS Teams.

Currently we build our solution on `document.referrer` but this is not persistent along the lifecycle of pop-up window.

I was looking at Teams SDK implementation today, the most suitable solution would be filling `window.opener.location` property when opening authorisation pop-up which is created by parent app. Since this we continue to different domain in authorisation flow, exposing this information should not create new security risk.

Knowing that we are in Teams created context would bring allow us to build more robust integration.

What are your thoughts?

**Comment by v-chetsh** [comment]
@marekcis - We will investigate on it and get back to you.

**Comment by Meghana-MSFT** [comment]
We are checking the redirection with Facebook authentication, We will update you soon.

**Comment by marekcis** [comment]
Just a friendly reminder about the progress here :) Thanks.

**Comment by Meghana-MSFT** [comment]
We have checked with [app-complete-auth](https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/app-complete-auth/csharp) sample which includes Facebook auth. We see that to reset password, it opens a popup and Facebook opens within that popup only. We then tried copying the link in popup and opened it in new browser window. We see that Facebook opens in the new window after resetting the password. So this implementation is by Google/Facebook etc. It will open within the new popup and this is not on the Teams side.
